### ISSUE TITLE
Build failed with python 3.7

### ISSUE THREAD

**Issue by niu541412** [healpy/healpy#473]
`healpy/src/_query_disc.cpp:9828:21: error: no member named 'exc_type' in '_ts'
    *type = tstate->exc_type;
            ~~~~~~  ^
healpy/src/_query_disc.cpp:9829:22: error: no member named 'exc_value' in '_ts'; did you mean 'curexc_value'?
    *value = tstate->exc_value;
                     ^~~~~~~~~
                     curexc_value
/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/pystate.h:237:15: note: 'curexc_value' declared here
    PyObject *curexc_value;
              ^
healpy/src/_query_disc.cpp:9830:19: error: no member named 'exc_traceback' in '_ts'; did you mean 'curexc_traceback'?
    *tb = tstate->exc_traceback;
                  ^~~~~~~~~~~~~
                  curexc_traceback
/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/pystate.h:238:15: note: 'curexc_traceback' declared
      here
    PyObject *curexc_traceback;
              ^
healpy/src/_query_disc.cpp:9837:24: error: no member named 'exc_type' in '_ts'
    tmp_type = tstate->exc_type;
               ~~~~~~  ^
healpy/src/_query_disc.cpp:9838:25: error: no member named 'exc_value' in '_ts'; did you mean 'curexc_value'?
    tmp_value = tstate->exc_value;
                        ^~~~~~~~~
                        curexc_value
/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/pystate.h:237:15: note: 'curexc_value' declared here
    PyObject *curexc_value;
              ^
healpy/src/_query_disc.cpp:9839:22: error: no member named 'exc_traceback' in '_ts'; did you mean 'curexc_traceback'?
    tmp_tb = tstate->exc_traceback;
                     ^~~~~~~~~~~~~
                     curexc_traceback
/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/pystate.h:238:15: note: 'curexc_traceback' declared
      here
    PyObject *curexc_traceback;
              ^
healpy/src/_query_disc.cpp:9840:13: error: no member named 'exc_type' in '_ts'
    tstate->exc_type = type;
    ~~~~~~  ^
healpy/src/_query_disc.cpp:9841:13: error: no member named 'exc_value' in '_ts'; did you mean 'curexc_value'?
    tstate->exc_value = value;
            ^~~~~~~~~
            curexc_value
/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/pystate.h:237:15: note: 'curexc_value' declared here
    PyObject *curexc_value;
              ^
healpy/src/_query_disc.cpp:9842:13: error: no member named 'exc_traceback' in '_ts'; did you mean 'curexc_traceback'?
    tstate->exc_traceback = tb;
            ^~~~~~~~~~~~~
            curexc_traceback
/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/pystate.h:238:15: note: 'curexc_traceback' declared
      here
    PyObject *curexc_traceback;
              ^
healpy/src/_query_disc.cpp:9897:24: error: no member named 'exc_type' in '_ts'
    tmp_type = tstate->exc_type;
               ~~~~~~  ^
healpy/src/_query_disc.cpp:9898:25: error: no member named 'exc_value' in '_ts'; did you mean 'curexc_value'?
    tmp_value = tstate->exc_value;
                        ^~~~~~~~~
                        curexc_value
/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/pystate.h:237:15: note: 'curexc_value' declared here
    PyObject *curexc_value;
              ^
healpy/src/_query_disc.cpp:9899:22: error: no member named 'exc_traceback' in '_ts'; did you mean 'curexc_traceback'?
    tmp_tb = tstate->exc_traceback;
                     ^~~~~~~~~~~~~
                     curexc_traceback
/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/pystate.h:238:15: note: 'curexc_traceback' declared
      here
    PyObject *curexc_traceback;
              ^
healpy/src/_query_disc.cpp:9900:13: error: no member named 'exc_type' in '_ts'
    tstate->exc_type = local_type;
    ~~~~~~  ^
healpy/src/_query_disc.cpp:9901:13: error: no member named 'exc_value' in '_ts'; did you mean 'curexc_value'?
    tstate->exc_value = local_value;
            ^~~~~~~~~
            curexc_value
/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/pystate.h:237:15: note: 'curexc_value' declared here
    PyObject *curexc_value;
              ^
healpy/src/_query_disc.cpp:9902:13: error: no member named 'exc_traceback' in '_ts'; did you mean 'curexc_traceback'?
    tstate->exc_traceback = local_tb;
            ^~~~~~~~~~~~~
            curexc_traceback
/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m/pystate.h:238:15: note: 'curexc_traceback' declared
      here
    PyObject *curexc_traceback;
              ^
1 warning and 15 errors generated.`


Maybe due to the cpp files built by the Cython 0.26 are not compatible with python 3.7. 
https://github.com/cython/cython/commit/8ad16fc871be075f889d5e7c2892db3e4c8ce978

**Comment by cjcopi** [comment]
I think your assessment is correct, a newer cython is needed to generate the appropriate .cpp files.

If you delete (from the unpacked healpy source directory)
```
healpy/src/_query_disc.cpp
healpy/src/_sphtools.cpp
healpy/src/_pixelfunc.cpp
```
then, during compile, these will be recreated using your version of cython.

A simple test from anaconda shows the following:
Python 3.6.6 : pypi source healpy builds
Python 3.7.0 : pypi source healpy fails as described above.

If I remove the stated .cpp files and have cython version 0.28.4 installed then
both versions build fine and pass the tests (`python3 setup.py test`).

I do not know if this is backwards compatible with older versions of python3 or with python2.

**Comment by MRCWirtz** [comment]
@cjcopi Worked for me, thank you!

**Comment by cdeil** [comment]
This is an issue that appears in all packages that use Cython and that ship Cython-generated C before Python 3.7 came out. It can be resolved just by making a new release with the latest stable Cython version and no other changes / fixes needed.

Would be very nice to have healpy with Python 3.7!

@zonca - I see you're already on it.

**Comment by zonca** [comment]
it worked, see #475 
I'll make another release.

**Comment by zonca** [comment]
can someone please verify if 1.12.4 works with Python 3.7? I used Cython 0.28.5

**Comment by cdeil** [comment]
I tried this on macOS:
```
conda create -n hp37 python=3.7 numpy astropy
pip install healpy
```
It doesn't work, there's plenty of compiler errors:
https://gist.github.com/cdeil/b4cc2758565bef7814073b174fccf1d5

Not sure if there's anything Python 3.7 specific, I didn't try e.g. Python 3.6. Usually I get the healpy binary from conda-forge, pip install healpy never worked reliably for me in the past years.

Over at https://github.com/conda-forge/healpy-feedstock/pull/16 I see this error
https://travis-ci.org/conda-forge/healpy-feedstock/jobs/420427492#L1990
and it seems a Python 3.7 build wasn't started by default, you have to add it somehow.

Also: why are you putting matplotib and scipy as required dependencies for healpy?
Is it really useful to install those every time someone asks for healpy to be installed?

https://github.com/healpy/healpy/blob/f6ce60ac554129a6270c0f7f37c8aadd86cc829d/setup.py#L521

**Comment by cjcopi** [comment]
In a Python 3.7 anaconda environment on Linux 
pip install healpy
worked like a charm.

**Comment by JulienPeloton** [comment]
To add to the discussion, I recently wanted to upgrade my version of healpy (python 3.6, macos):
```bash
pip install --user --upgrade healpy
```
and get the same `fatal error: 'cstdint' file not found`. Any update on this?

Thanks!

**Comment by zonca** [comment]
I don't have a macos platform (and neither any experience with macos) to debug this. if someone finds a solution please create a pull request and I can review and merge it. Then I should also be able to build macos conda packages.

**Comment by pkgw** [comment]
The 1.12.6 release seems to have put together with a too-old version of Cython, so that the distributed files are incompatible with Python 3.7 again.

**Comment by zonca** [comment]
my bad, I released 1.12.7 to fix this, please reopen if the issue appears again
